app/services/parameters_fetch_service.rb

```diff
@@ -0,0 +1,84 @@
+# frozen_string_literal: true
+
+class ParametersFetchService
+  def self.clone_or_pull_git_repo
+    raise if clone_url.blank?
+
+    if File.directory?(git_clone_folder)
```

Comment:
could this be more DRY by moving the git cloning to one place, and calling it from the two fetch services (scenarios and parameters)

Reply:
@Br3nda Yup I agree. Will do in next PR